### PR TITLE
detect toyota tss-p canfilter device as smartDsu to enable long

### PR DESCRIPTION
**Pull request by Smartype**

### CODE REVIEW

**Comment by cydia2020** [comment]
Are we able to add these cars to `NO_STOP_TIMER_CAR`, discord users are saying their cars can automatically resume after coming to a complete stop if the standstill bit doesn't get touched.

File: selfdrive/car/toyota/interface.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
     ret.enableBsm = 0x3F6 in fingerprint[0] and candidate in TSS2_CAR
 
     # Detect smartDSU, which intercepts ACC_CMD from the DSU (or radar) allowing openpilot to send it
-    if 0x2FF in fingerprint[0]:
+    if 0x2FF in fingerprint[0] or 0x2AA in fingerprint[0]:
```
Comment (sshane) [inline selfdrive/car/toyota/interface.py:207]: is this a different device from the smart DSU? the comment says it already works with the radar

Comment (cydia2020) [reply selfdrive/car/toyota/interface.py:207]: It's a different hardware/firmware with a lot more redundancy initially but the PR was rejected. #24695

Comment (williefeb19) [reply selfdrive/car/toyota/interface.py:207]: This version is also cheaper as it is not based on a panda.

Comment (sshane) [reply selfdrive/car/toyota/interface.py:207]: can you use the SDSU on these cars? this hardware just enables more features like changing ACC_TYPE?

Comment (cydia2020) [reply selfdrive/car/toyota/interface.py:207]: > can you use the SDSU on these cars? this hardware just enables more features like changing ACC_TYPE?

technically, yea, if you can find the right connector / waterproofing for it.

0x2AA is a copy of the stock 0x343 signal, which also passes through various stock ACC-related states like distance button, lead state, permit braking, etc.

In the original PR for opendbc:
`BO_ 682 ACC_CONTROL_COPY: 8 XXX`

Comment (Smartype) [reply selfdrive/car/toyota/interface.py:207]: Let me explain why I didn't use SDSU and try to create this device(firmware) at the first place. 

1. I can get find a connector for the can gateway so I have to install this device in the engine bay. So it has to be small in size, water proof, easy wiring. 

2. The board is the well know MB CAN filter, a mileage manipulator. Which is very easy to get on eBay,Aliexpress. It has two CAN buses, 12V input. Perfect for our purpose and easy to wire. 

3. One of the feature which is vey important is update FW over can bus. 


> can you use the SDSU on these cars? this hardware just enables more features like changing ACC_TYPE?

Comment (sshane) [reply selfdrive/car/toyota/interface.py:207]: We have 0x2FF in the DBC, what does 0x2AA publish?

Comment (cydia2020) [reply selfdrive/car/toyota/interface.py:207]: > `BO_ 682 ACC_CONTROL_COPY: 8 XXX`

0x2AA is a copy of 0x343

**Comment by sshane** [comment]
I think if we added a route to test_models (https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py), this should be fine to merge, it's very similar to https://github.com/commaai/openpilot/pull/28440, we're just allowing another piece of hardware to act as the SDSU. @adeebshihadeh let me know if you have any objections.

**Comment by sshane** [comment]
@Smartype closing for now until you have a public test route we can use.

**Comment by cydia2020** [comment]
https://connect.comma.ai/2f37c007683e85ba/1693683584155/1693684571551

**Comment by Smartype** [comment]
> @Smartype closing for now until you have a public test route we can use.

Can you please reopen so that I can update the PR to include the test route? Thanks.

**Comment by sshane** [comment]
https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c

**Comment by Smartype** [comment]
1. Write down the current commit hash of your PR-branch git log --oneline -1 <PR-BRANCH>
2. Write down the latest commit hash on github before the PR has been closed.
3. git push -f origin <GITHUB-HASH-FROM-STEP-2>:<PR-BRANCH>
4. Reopen the PR.
5. git push -f origin <HASH-FROM-STEP-1>:<PR-BRANCH>

@sshane  I have done 1,2,3 to switch the branch to the state before closing. Please reopen.(step 4)

**Comment by sshane** [comment]
Commit still looks wrong.

**Comment by Smartype** [comment]
> Commit still looks wrong. 

Oh I have rebased the branch to sync with latest master. Let me try if I can restore it to the state before closing.

**Comment by Smartype** [comment]
@sshane Now the latest commit id matches with this PR. please advice.

**Comment by Smartype** [comment]
Hi, @sshane I have added the test route. Do I have to rebase the branch to latest master?

**Comment by sshane** [comment]
looks like the TSS-P Camry has responsive enough control!

image

**Pull request merged**